web/src/components/dashboard/SidebarCardFundingNudge.tsx

```diff
@@ -13,16 +23,104 @@ type Props = {
 
 export const SidebarCardFundingNudge = (props: Props): ReactElement => {
   const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const [profileData, setProfileData] = useState<ProfileData>(createEmptyProfileData());
+
+  const { queueUpdateTaskProgress } = useUpdateTaskProgress();
+  const close = (): void => {};
+  const onSuccess = (): void => {};
   const router = useRouter();
   const { updateQueue, business } = useUserData();
 
-  const updateToUpAndRunningAndCompleteTaxTask = async (): Promise<void> => {
+  useMountEffectWhenDefined(() => {
     if (!business) return;
+    setProfileData(business.profileData);
+  }, business);
+
+  const displayBusinessName = (): boolean => {
+    return LookupLegalStructureById(business?.profileData.legalStructureId).elementsToDisplay.has(
+      "businessName"
+    );
+  };
+
+  const displayResponsibleOwnerName = (): boolean => {
+    return LookupLegalStructureById(business?.profileData.legalStructureId).elementsToDisplay.has(
+      "responsibleOwnerName"
+    );
+  };
+
+  const updateToUpAndRunningAndCompleteTaxTask = async (): Promise<void> => {
+    if (!business || !updateQueue) return;
     await updateQueue
       ?.queueProfileData({
         operatingPhase: OperatingPhaseId.UP_AND_RUNNING,
       })
       .update();
+
+    const encryptedTaxId =
+      profileData.taxId === business.profileData.taxId ? profileData.encryptedTaxId : undefined;
+
+    try {
+      let businessNameToSubmitToTaxApi = "";
+
+      if (displayBusinessName()) {
+        businessNameToSubmitToTaxApi = profileData.businessName;
+      }
+      if (displayResponsibleOwnerName()) {
+        businessNameToSubmitToTaxApi = profileData.responsibleOwnerName;
+      }
+
+      if (!businessNameToSubmitToTaxApi) {
+        return;
+      }
+
+      const userDataToSet = await postTaxFilingsOnboarding({
+        taxId: profileData.taxId as string,
+        businessName: businessNameToSubmitToTaxApi,
+        encryptedTaxId: encryptedTaxId as string,
+      });
+
+      updateQueue.queue(userDataToSet).queueProfileData({
+        taxId: profileData.taxId,
+        encryptedTaxId: encryptedTaxId,
```

Comment:
dust - I'm not sure I understand whats going on here exactly. Can you explain it further?